rpc/codec.go

```diff
@@ -9,5 +9,6 @@ type Codec interface {
 	EncodeBytes(interface{}) ([]byte, error)
 	Encode(interface{}, io.Writer) error
 	DecodeBytes(interface{}, []byte) error
+	DecodeBytesPtr(interface{}, []byte) error
```

Comment:
I'm not sure we need this. I think it will work with `DecodeBytes`, did you try that?